core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -886,12 +890,20 @@ static boolean shouldIgnoreOrRejectNewClusterState(ESLogger logger, ClusterState
         }
     }
 
-    private void handleJoinRequest(final DiscoveryNode node, final MembershipAction.JoinCallback callback) {
+    void handleJoinRequest(final DiscoveryNode node, final MembershipAction.JoinCallback callback) {
```

Comment:
oops... no, because it is used in a test (ZenDiscoveryTests#testHandleNodeJoin_incompatibleMinVersion)